### PR TITLE
Slicing integration tests

### PR DESCRIPTION
**Pull request by vincentschen**
Fix integration test for slicing convergence (previous version was broken)

**Test plan**: Added integration tests for slicing initialization + training.

### CODE REVIEW

File: test/slicing/test_convergence.py [hunk 1]
```diff
@@ -14,31 +16,42 @@
...
+N_TRAIN = 1500
+N_VALID = 300
 
 
-class SlicingTest(unittest.TestCase):
+class SlicingConvergenceTest(unittest.TestCase):
     @classmethod
     def setUpClass(cls):
-        cls.trainer_config = {"n_epochs": 3, "progress_bar": False}
+        cls.trainer_config = {
```
Comment (henryre) [inline test/slicing/test_convergence.py]: Can we just pass these directly to the `train` call? More readable that way

File: test/slicing/test_convergence.py [hunk 2]
```diff
@@ -14,31 +16,42 @@
...
 def g(x: DataPoint) -> int:
-    return 1 if x.x1 > x.x2 + 0.3 else 0
+    radius = 0.3
+    h, k = (0.25, 0.0)  # center
+    return np.sqrt((x.x1 - h) ** 2 + (x.x2 - k) ** 2) < radius
 
 
-N_TRAIN = 1000
-N_VALID = 100
+SEED = 123
```
Comment (henryre) [inline test/slicing/test_convergence.py]: This isn't reused, just put in set_seed call

File: test/slicing/test_convergence.py [hunk 3]
```diff
@@ -61,36 +73,49 @@ def test_slicing(self):
...
 
         # Add slice labels
-        add_slice_labels(dataloaders[0], task1, S_train, slice_names)
-        add_slice_labels(dataloaders[1], task1, S_valid, slice_names)
+        add_slice_labels(dataloaders[0], base_task, S_train, slice_names)
+        add_slice_labels(dataloaders[1], base_task, S_valid, slice_names)
+
+        # Test that slice labels have been added
+        Y_dict_train = dataloaders[0].dataset.Y_dict
+        self.assertIn("task", Y_dict_train.keys())
```
Comment (henryre) [inline test/slicing/test_convergence.py]: Do we need the keys() calls for these?

File: test/slicing/test_convergence.py (hunk 2 above)
Comment (vincentschen) [reply test/slicing/test_convergence.py]: see update, now it's used in both tests

File: test/slicing/test_convergence.py (hunk 3 above)
Comment (vincentschen) [reply test/slicing/test_convergence.py]: nope! noted— but moving to unit tests

File: test/slicing/test_convergence.py [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+        self.assertGreater(scores["task/TestData/valid/accuracy"], 0.95)
+        self.assertGreater(scores["task_slice:h_pred/TestData/valid/accuracy"], 0.95)
+        self.assertGreater(scores["task_slice:h_ind/TestData/valid/f1"], 0.95)
+
+        # Calculate/check train/val loss
+        train_dataset = dataloaders[0].dataset
+        train_loss_output = model.calculate_loss(
+            train_dataset.X_dict, train_dataset.Y_dict
+        )
+        train_loss = float(train_loss_output[0]["task"].detach().numpy())
```
Comment (henryre) [inline test/slicing/test_convergence.py]: nit: `.item()`

File: test/slicing/test_slice_combiner.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+
+class SliceCombinerTest(unittest.TestCase):
+    @classmethod
+    def setup_class(cls):
+        set_seed(123)
+
+    def test_init_and_forward(self):
+        batch_size = 4
+        h_dim = 20
```
Comment (henryre) [inline test/slicing/test_slice_combiner.py]: nit: I think you meant to use this in task_slice:base_pred_transform

**Pull request merged**